Source/codec.cpp

```diff
@@ -87,52 +87,40 @@ DWORD codec_get_encoded_len(DWORD dwSrcBytes)
 	return dwSrcBytes + 8;
 }
 
-void codec_encode(void *pbSrcDst, int size, int size_64, char *pszPassword)
+void codec_encode(BYTE* pbSrcDst, DWORD size, int size_64, char *pszPassword)

```

Comment:
Can you tell please, how have you decided to change those two types?

Reply:
First one was always casted to `char*` before, but technically function is called only with `BYTE*` arrays currently. For assembly I think it doesn't matter in this case if it's signed or unsiged. But `void*` is almost useless and always needs to be casted so I don't think it should be used in this case.

As for `size` I think it shows in comparison instructions, in this function changing size to signed makes one insturction `jl` instead of `jb`, manual cast can be used at a specific place but it doesn't look nice. In `codec_decode` presence of unsigned arithmetic is even more obvious since remainder of division by `64` is replaced with `and` instruction.